lib/spack/spack/schema/packages.py

```diff
@@ -21,6 +21,20 @@
                 "default": {},
                 "additionalProperties": False,
                 "properties": {
+                    "require": {
+                        "oneOf": [
+                            # A condition group: at least one member is applied
+                            {
+                                "type": "array",
+                                "items": {
+                                    "type": "object",
+                                    "properties": {"one_of": {"type": "array"}},
```

Comment:
As mentioned in https://github.com/spack/spack/pull/27987#issuecomment-1204396111 I would either go with:
```suggestion
                                    "properties": {"any_of": {"type": "array"}},
```
or change the semantics to "exactly one of".

Reply:
If the bugfix below works, it might be easier to change to "exactly one of".

Reply:
Also, if we're using "exactly one of" we'll have less trouble with requirement weights (it will be one weight per group per member)